src/bicgstab_kernel.cu

```diff
@@ -1971,3 +1971,37 @@ __global__ void PP_BC_p_T(real *A, int pitch, dom_struct *dom)
   }
 }
 */
+
+__global__ void coeffs_refine(dom_struct *dom, int pitch, real *values,
+  int *phase, int *offsets, real *rhs)
+{
+  int i, j;  // iterator
+  int C, CC, CCC;  // cell
+  int jId;  // column id of nonzero elements
+  int ii, jj, kk;
+
+  int tj = blockIdx.x * blockDim.x + threadIdx.x;
+  int tk = blockIdx.y * blockDim.y + threadIdx.y;
+
+  if(tj < dom->Gcc.jn && tk < dom->Gcc.kn) {
+        for(i = dom->Gcc.is-DOM_BUF; i < dom->Gcc.ie-DOM_BUF; i++) {
```

Comment:
Are these loop indices correct? I'm more used to `for (i = is; i < ie; i++)` and then `C = i - DOM_BUF`. I guess they're the same.